test/priorusetest.jl

```diff
@@ -0,0 +1,83 @@
+
+using IncrementalInference
+using Test
+@testset "Test priors" begin
+
+    N=200
+    graphinits = [false, true]
+    for  graphinit = graphinits
+
+        fg = initfg()
+        fg.solverParams.N = N
+
+
+        addVariable!(fg, :x0, ContinuousScalar, autoinit = graphinit, N=N)
+        addFactor!(fg, [:x0], Prior(Normal(-1.0, 1.0)))
+
+        addVariable!(fg, :x1, ContinuousScalar, autoinit = graphinit, N=N)
+
+        addVariable!(fg, :x2, ContinuousScalar, autoinit = graphinit, N=N)
+        addFactor!(fg, [:x2], Prior(Normal(+1.0, 1.0)))
+
+        addFactor!(fg, [:x0; :x1], LinearConditional(Normal(0.0, 0.1)))
+        addFactor!(fg, [:x1; :x2], LinearConditional(Normal(0.0, 0.1)))
+
+        #solve
+        tree, smt, hist = solveTree!(fg)
+        x0_m = getKDEMean(getKDE(getVariable(fg, :x0)))[1]
+        x1_m = getKDEMean(getKDE(getVariable(fg, :x1)))[1]
+        x2_m = getKDEMean(getKDE(getVariable(fg, :x2)))[1]
+
+        @info ("Testing means = 0 with 2 priors:\ngraphinit=$graphinit\nMeans: x0: $(x0_m), x1: $x1_m, x2: $x2_m")
+
+        @test isapprox(x0_m, 0.0, atol = 0.1)
+        @test isapprox(x1_m, 0.0, atol = 0.1)
+        @test isapprox(x2_m, 0.0, atol = 0.1)
```

Comment:
@Affie, first question: if you are putting two priors at the ends of a 3 variable chain `(x0,x1,x2)`, either at + or - 1 -- why would you expect them to all be at average 0?  I would expect them to be at + or 1 one respectively, with `x1` floating somewhere 0 in the middle?

This was based on me reading the code.

Reply:
x0 and x2 should be near 0, and only x1 should be "on" 0.  Test with loose `atol=0.1` bit hacky, but okay :-)  x0 and x2 should be around 0.1 or something like that.

Reply:
Please note, I changed the standard deviation to 0.01.
So its more like -0.01, 0, 0.01 for x0,x1,x2.

-1/1 <-x0-> 1/0.01 <-x1-> 1/0.01 <-x2-> 1/1

Reply:
i'm assuming you did the calculation for multiplying two Gaussians, and found that the new mean should be at +- 0.01?  What I mean is if you were to multiply N(-1,0.1) * N(+1,0.1), the answer will be at zero with a new covariance slightly smaller than 0.1.